src/pages/QuestionPage.tsx

```diff
@@ -103,17 +112,39 @@ export default function QuestionPage() {
     }
   }
 
-  const nextQuestion = (event:React.MouseEvent) => {
+  const nextQuestion = (event: React.MouseEvent) => {
     // Submit answer to backend
-    if (!requestNextQuestionButtonPressed){
+    if (!requestNextQuestionButtonPressed) {
       console.log("Next question button pressed");
       dispatch(setRequestNextQuestion(true));
       setRequestNextQuestionButtonPressed(true);
+      // reset the leaderboard for the next question
+      dispatch(resetLeaderboard());
+      dispatch(setRequestUpdatedLeaderboard(false));
```

Comment:
This line is not needed. You already set the flag `requestUpdatedLeaderboard` to `false` [here](https://github.com/Jhappy77/cpsc559-project-front-end/blob/76499c9ef64986de938fbd37f482f3f40383cc8a/src/state/leaderboardSlice.ts#L28). You could either remove this line or remove [this](https://github.com/Jhappy77/cpsc559-project-front-end/blob/76499c9ef64986de938fbd37f482f3f40383cc8a/src/state/leaderboardSlice.ts#L28) line. If you feel that everytime leaderboard is reset that `requestUpdatedLeaderboard` needs to be set to false every time, you should remove this line.